lib/ex_unit/test/ex_unit/doc_test_test.exs

```diff
@@ -200,6 +200,62 @@ defmodule ExUnit.DocTestTest.IndentationNotEnough do
   def test_fun, do: :ok
 end |> write_beam
 
+defmodule ExUnit.DocTestTest.FencedHeredocs do
+  @doc ~S'''
+  Receives a test and formats its failure.
+
+  ## Examples
+
+  ```
+  iex> 1 + 2
+  3
+  ```
+
+      ```
+      iex> 1 + 2
+      3
+      ```
+
+  ```
+      iex> 1 + 2
+      3
+  ```
+  '''
+  def heredocs, do: :ok
+end |> write_beam
+
+defmodule ExUnit.DocTestTest.FencedIndentationMismatchedPrompt do
```

Comment:
I don't think we need the test cases from now on below. They are all tested without fencing and I don't see why it is required to test them again since fencing does not affect them.